src/osd/OSD.h

```diff
@@ -1631,8 +1631,7 @@ class OSD : public Dispatcher,
   map<int, int> debug_heartbeat_drops_remaining;
   Cond heartbeat_cond;
   bool heartbeat_stop;
-  Mutex heartbeat_update_lock; // orders under heartbeat_lock
-  bool heartbeat_need_update;   ///< true if we need to refresh our heartbeat peers
+  std::atomic_bool heartbeat_need_update;   
```

Comment:
@tchaikov, I tried std::atomic_flag, but I still prefer std::atomic<>. The reason is below:
1) 
#include <iostream>
#include <utility>
#include <atomic>

int main()
{
    std::atomic<bool> lock;
    std::cout << "free or not:" << std::endl;
    std::cout << lock.is_lock_free() << std::endl;
    return 0;
}

**Output:**
free or not:
1

So std::atomic<bool> is lock free.

2) there is not a function load for atomic_flag, so I have to do some change for heartbeat_peers_need_update return value and implement.

Do you agree?